File: src/pixelHat/InvadersMovie.java
package pixelHat;

import java.io.PrintStream;

public class InvadersMovie {

	public static void main(String[] args) throws Exception {
		
		SpriteLoader sprites = new SpriteLoader();
				
		int [] invaderMap = {'#', 1};
		int [][][] invaders = {
			sprites.doubler(sprites.colorSprite("invaderA1", invaderMap)),
			sprites.doubler(sprites.colorSprite("invaderA2", invaderMap)),	
			sprites.doubler(sprites.colorSprite("invaderB1", invaderMap)),	
			sprites.doubler(sprites.colorSprite("invaderB2", invaderMap)),	
			sprites.doubler(sprites.colorSprite("invaderC1", invaderMap)),	
			sprites.doubler(sprites.colorSprite("invaderC2", invaderMap))
		};
				
		PrintStream ps = new PrintStream("invGEN.txt");
		
		for(int y=-15;y<0;++y) {
			HatFrame f = new HatFrame();
			int ofs = y%2;
			if(ofs<0) ofs=-ofs;
			f.drawSprite(0, y, invaders[2+ofs]);	
			f.drawSprite(35,y, invaders[4+ofs]);
			ps.println("%");
			ps.println(f.toString());
		}
		
		
		for(int x=0;x<64;++x) {
			HatFrame f = new HatFrame();
			int ofs = x%2;
			if(ofs<0) ofs=-ofs;
			f.drawSprite(x+0, 0, invaders[2+ofs]);	
			f.drawSprite(x+35,0, invaders[4+ofs]);
			ps.println("%");
			ps.println(f.toString());
		}
		
		for(int y=0;y>-16;--y) {
			HatFrame f = new HatFrame();
			int ofs = y%2;
			if(ofs<0) ofs=-ofs;
			f.drawSprite(0, y, invaders[2+ofs]);	
			f.drawSprite(35,y, invaders[4+ofs]);
			ps.println("%");
			ps.println(f.toString());
		}
				
		ps.flush();
		ps.close();

	}

}

File: movieMaker/InvadersMovie.py
import SpriteLoader
import HatFrame
from errno import EROFS

sprites = SpriteLoader.SpriteLoader()

invaderMap = ['#', 1]
invaders = [
    sprites.doubler(sprites.colorSprite("invaderA1", invaderMap)),
    sprites.doubler(sprites.colorSprite("invaderA2", invaderMap)),    
    sprites.doubler(sprites.colorSprite("invaderB1", invaderMap)),    
    sprites.doubler(sprites.colorSprite("invaderB2", invaderMap)),    
    sprites.doubler(sprites.colorSprite("invaderC1", invaderMap)),    
    sprites.doubler(sprites.colorSprite("invaderC2", invaderMap))
    ]

with open("invGEN.txt","w") as ps:
    
    for y in range(-15,0,1):    
        f = HatFrame.HatFrame()
        ofs = y%2
        if ofs<0:
             ofs=-ofs
        f.draw_sprite(0, y, invaders[2+ofs])    
        f.draw_sprite(35,y, invaders[4+ofs])
        ps.write("%\n")
        ps.write(f.to_string()+"\n")

    
    for x in range(0,64):
        f = HatFrame.HatFrame()
        ofs = x%2
        if ofs<0:
            ofs=-ofs
        f.draw_sprite(x+0, 0, invaders[2+ofs]);    
        f.draw_sprite(x+35,0, invaders[4+ofs]);
        ps.write("%\n")
        ps.write(f.to_string()+"\n")
       
    for y in range(0,-16,-1):
        f = HatFrame.HatFrame()
        ofs = y%2
        if ofs<0:
            ofs=-ofs
        f.draw_sprite(0, y, invaders[2+ofs]);    
        f.draw_sprite(35,y, invaders[4+ofs]);
        ps.write("%\n")
        ps.write(f.to_string()+"\n")
        